bpf/protocol_http.h

```diff
@@ -34,6 +34,110 @@ static __always_inline http_info_t *empty_http_info() {
     return value;
 }
 
+static __always_inline u32 trace_type_from_meta(http_connection_metadata_t *meta) {
+    if (meta->type == EVENT_HTTP_CLIENT) {
+        return TRACE_TYPE_CLIENT;
+    }
+
+    return TRACE_TYPE_SERVER;
+}
+
+static __always_inline void http_get_or_create_trace_info(http_connection_metadata_t *meta,
+                                                          u32 pid,
+                                                          connection_info_t *conn,
+                                                          void *u_buf,
+                                                          int bytes_len,
+                                                          s32 capture_header_buffer) {
+    tp_info_pid_t *tp_p = tp_buf();
+
+    if (!tp_p) {
+        return;
+    }
+
+    tp_p->tp.ts = bpf_ktime_get_ns();
+    tp_p->tp.flags = 1;
+    tp_p->valid = 1;
+    tp_p->pid = pid; // used for avoiding finding stale server requests with client port reuse
+    tp_p->req_type = (meta) ? meta->type : 0;
+
+    urand_bytes(tp_p->tp.span_id, SPAN_ID_SIZE_BYTES);
+
+    u8 found_tp = 0;
+
+    if (meta) {
+        if (meta->type == EVENT_HTTP_CLIENT) {
+            pid_connection_info_t p_conn = {.pid = pid};
+            __builtin_memcpy(&p_conn.conn, conn, sizeof(connection_info_t));
+            found_tp = find_trace_for_client_request(&p_conn, &tp_p->tp);
+        } else {
+            //bpf_dbg_printk("Looking up existing trace for connection");
+            //dbg_print_http_connection_info(conn);
```

Comment:
Just double checking you intended to leave those in.

Reply:
Yeah it's good for the future if we need to debug the code.